PlayRho/Common/Math.hpp

```diff
@@ -642,7 +642,7 @@ template <typename T>
 constexpr auto GetModuloNext(const T value, const T count) noexcept -> T
 {
     assert(value < count);
-    return (value + 1) % count; // NOLINT(clang-analyzer-core.DivideZero)
+    return (unsigned char) (value + 1) % (unsigned char) count; // NOLINT(clang-analyzer-core.DivideZero)
```

Comment:
I don't think casting an arbitrary type `T` unconditionally to `unsigned char` will do the right thing except for when `T` is itself only an `unsigned char` - which is may be the only way this function template is actually used by the library but given that this is a function template it needs to work properly for any type supporting modulo and increment by one. I have places like this where I haven't setup the templates as constrained as I'd prefer and have been meaning to review all generic code for that purpose.

I also don't want to have any C-style casts in any of the library code.

Reply:
My bad this was a test. Nonetheless, instead of using % operator, it may be better to use [modulus](https://en.cppreference.com/w/cpp/utility/functional/modulus) (I never use it but it seems to be more in accordance with the code)